### PR TITLE
remove the data package cruft

### LINKED ISSUE #1211: Remove residual datapackage infrastructure
Now that the SQLite and Parquet based ETL is working (#1176) and using the new metadata structure, all of the residual datapackage infrastructure can be removed. The data packages have a ✨ special place ✨ in @cmgosnell's 💖 and so she gets the cathartic honors and the right to tell @zaneselvans I Told You So at least 768 more times. Because she did.

# 💀  🪓 🔥 The Datapackage is Dead. 🔥 🪓 💀

## Stuff that can go now
* [x] 🪓 Remove `pudl.convert.datapkg_to_rst` (became `metadata_to_rst`)
* [x] 🌈 Update dependencies in `setup.py` to remove anything that we don't need anymore because we aren't outputting datapackages (but we still use them in the datastore).
* [x] 🪓 The `pudl.load.csv` module
* [x] 🔥 The `pudl.load.metadata` module
* [x] 💀 The `pudl.convert.merge_datapkgs` module
* [x] ☠️ The `pudl.convert.datapkg_to_sqlite` modules
* [x] 🔥 Remove `src/pudl/package_data/meta/datapkg/datapackage.json`

## Don't remove
* 🛑 ✋🏼 ⚠️ `pudl.convert.epacems_to_parquet.py` has been repurposed as a script that allows you to run the EPA CEMS ETL separate from (after) the SQLite based FERC/EIA ETL, which is a convenient feature to have around.

### PR DESCRIPTION
**Pull request by cmgosnell**
Closes #1211

### CODE REVIEW

**Comment by zaneselvans** [comment]
I think there are some additional references to some of these modules in `.pre-commit-config.yaml` and `.coveragerc` and `docs/conf.py` that should be removed as well.

File: docs/conf.py [hunk 1]
```diff
@@ -59,12 +59,7 @@
 autoapi_type = 'python'
 autoapi_dirs = ['../src/pudl', ]
 autoapi_ignore = [
-    "*/convert/datapkg_to_sqlite.py",
-    "*/convert/merge_datapkgs.py",
-    "*/load/csv.py",
-    "*/load/metadata.py",
     "*_test.py",
-    "*/package_data/*",
```
Comment (zaneselvans) [inline docs/conf.py:67]: I think we do need to keep this line actually -- we don't want AutoAPI trying to build docs from stuff under that directory.

**Review by zaneselvans — CHANGES_REQUESTED** [review]
One tiny oops in there. but mostly 🎉 yay 🎉

File: docs/conf.py (hunk 1 above)
Comment (cmgosnell) [reply docs/conf.py:67]: lol i was a little trigger happy!

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**